### PR TITLE
LookupResolver: Implement LookupSRV()

### PR DESCRIPTION
**Pull request by karasz**

### CODE REVIEW

File: resolver_srv.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	if service == "" && proto == "" {
+		target = name
+	} else {
+		target = "_" + service + "._" + proto + "." + name
+	}
+	target = dns.Fqdn(target)
+
+	msg, err := r.h.Lookup(ctx, dns.CanonicalName(target), dns.TypeSRV)
+	if err != nil {
```
Comment (amery) [inline resolver_srv.go:24]: not calling validateResp() ?

File: resolver_srv.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		return "", nil, err
+	}
+	ForEachAnswer(msg, func(rr *dns.SRV) {
+		netsrv = append(netsrv, makeNetSRV(rr))
```
Comment (amery) [inline resolver_srv.go:28]: this could lead to appending nil entries

File: resolver_srv.go (hunk 1 above)
Comment (karasz) [reply resolver_srv.go]: solved

File: resolver_srv.go (hunk 2 above)
Comment (karasz) [reply resolver_srv.go]: not anymore

**Comment by amery** [comment]
Looks good :)

**Review by amery — APPROVED** [review]

**Comment by amery** [comment]
v0.4.12 tagged

**Pull request merged**